stubs/openpyxl/openpyxl/worksheet/dimensions.pyi

```diff
@@ -1,95 +1,118 @@
-from _typeshed import Incomplete
+from _typeshed import Incomplete, Unused
+from collections.abc import Callable, Generator
+from typing import Generic, TypeVar
+from typing_extensions import Self
 
 from openpyxl.descriptors import Strict
+from openpyxl.descriptors.base import Alias, Bool, Float, Integer, String
 from openpyxl.descriptors.serialisable import Serialisable
 from openpyxl.styles.styleable import StyleableObject
 from openpyxl.utils.bound_dictionary import BoundDictionary
+from openpyxl.worksheet.worksheet import Worksheet
+from openpyxl.xml.functions import Element
+
+_DimT = TypeVar("_DimT", bound=Dimension)
 
 class Dimension(Strict, StyleableObject):
-    __fields__: Incomplete
-    index: Incomplete
-    hidden: Incomplete
-    outlineLevel: Incomplete
-    outline_level: Incomplete
-    collapsed: Incomplete
-    style: Incomplete
+    __fields__: tuple[str, ...]
+
+    index: Integer
+    hidden: Bool
+    outlineLevel: Integer
+    outline_level: Alias
+    collapsed: Bool
+    style: Alias
+
     def __init__(
-        self, index, hidden, outlineLevel, collapsed, worksheet, visible: bool = ..., style: Incomplete | None = ...
+        self,
+        index: int,
+        hidden: bool,
+        outlineLevel: int | None,
+        collapsed: bool,
+        worksheet: Worksheet,
+        visible: bool = True,
+        style: Incomplete | None = None,
     ) -> None: ...
-    def __iter__(self): ...
-    def __copy__(self): ...
+    def __iter__(self) -> Generator[tuple[str, str], None, None]: ...
+    def __copy__(self) -> Self: ...
 
 class RowDimension(Dimension):
-    __fields__: Incomplete
-    r: Incomplete
-    s: Incomplete
-    ht: Incomplete
-    height: Incomplete
-    thickBot: Incomplete
-    thickTop: Incomplete
+    __fields__: tuple[str, ...]
+
+    r: Alias
+    s: Alias
+    ht: Float
+    height: Alias
+    thickBot: Bool
+    thickTop: Bool
```

Comment:
How useful are these `Alias` annotations? We don't currently have a return-type annotation for `openpyxl.descriptors.base.Alias.__get__`. But even if we did, I think the only accurate return type we could give it would be `Any`, since there's no way of expressing statically "this is a descriptor for aliasing an attribute to another arbitrary attribute" (unless we made `Alias` generic in the stubs, I suppose).

https://github.com/python/typeshed/blob/f6f1fa3df4d41d5267a4440204b3d0d3c198e9ed/stubs/openpyxl/openpyxl/descriptors/base.pyi#L71-L75

We might be better off using simpler annotations for now, which will accurately reflect the types users will get at runtime if they access these attributes on instances of `RowDimension`:

```suggestion
    r: int
    s: int
    ht: Float
    height: float
    thickBot: Bool
    thickTop: Bool
```

Reply:
Making `Alias` generic is probably the best course of action, although I'd consider this out of scope for this PR.

Reply:
Okay, we can leave this for now